File: ios/Plugin/LockScreenOrientationPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(LockScreenOrientationPlugin, "LockScreenOrientation",
           CAP_PLUGIN_METHOD(echo, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(lockOrientationToLandscape, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(lockOrientationToPortrait, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(unlockOrientation, CAPPluginReturnPromise);
)

File: ios/Plugin/LockScreenOrientationPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(LockScreenOrientationPlugin)
public class LockScreenOrientationPlugin: CAPPlugin {
    private let implementation = LockScreenOrientation()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }

    @objc func lockOrientationToLandscape(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve()
    }

    @objc func lockOrientationToPortrait(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve()
    }

    @objc func unlockOrientation(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve()
    }
}
